app/controllers/changes_controller.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+class ChangesController < ApplicationController
+  def undo
+    version = PaperTrail::Version.find(params.require :id)
+    head :unauthorized and return unless version.done_by? @current_user
+    # Reify only returns false when the thing didn't exist beforehand.
+    if version.reify
+      version.reify.save!
```

Comment:
They can (see the undo comment I put on generating rotations), but those would be multiple versions anyway.